### PR TITLE
Fix for a recent regression to develop branch testing

### PR DESCRIPTION
**Pull request by jkryl**
The current code works but in case that develop should not be
tested and error() is called in init stage, the final result
is ERROR instead of SUCCESS. This fix hasn't been tested but
according to stack-overflow it should work ;-)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -59,13 +59,18 @@ String cron_schedule = BRANCH_NAME == "develop" ? "0 2 * * *" : ""
 
 pipeline {
   agent none
+  options {
+    timeout(time: 2, unit: 'HOURS')
```
Comment (bobek) [inline Jenkinsfile:63]: :+1:

File: Jenkinsfile [hunk 2]
```diff
@@ -76,8 +81,9 @@ pipeline {
           if (currentBuild.getBuildCauses('jenkins.branch.BranchIndexingCause') &&
               BRANCH_NAME == "develop") {
               print "INFO: Branch Indexing, aborting job."
-              currentBuild.result = 'ABORTED'
-              error('Stopping early')
+              currentBuild.getRawBuild().getExecutor().interrupt(Result.SUCCESS)
+              sleep(3)   // Interrupt is not blocking and does not take effect immediately.
```
Comment (bobek) [inline Jenkinsfile:85]: The magic sleep ...  the glue of the Internet. How safe is 3 (I assume seconds)? The whole pipeline takes long time, so we can also do 10s ;)

**Review by bobek — APPROVED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (jkryl) [reply Jenkinsfile:85]: yea :-) The example I took this from had 1 second so I thought that 3 should be enough, but who knows :-)

**Review by tiagolobocastro — APPROVED** [review]

**Comment by jkryl** [comment]
bors merge

**Pull request merged**